jwst/pixel_replace/pixel_replace.py

```diff
@@ -287,32 +314,55 @@ def fit_profile(self, model):
             adjacent_inds = set(range(ind - self.pars['n_adjacent_cols'], ind + self.pars['n_adjacent_cols'] + 1))
             adjacent_inds.discard(ind)
             valid_adjacent_inds = list(adjacent_inds.intersection(valid_profiles))
+
             # Cut out valid neighboring profiles
-            profile_data = model.data[self.custom_slice(dispaxis, valid_adjacent_inds)]
+            adjacent_condition = self.custom_slice(dispaxis, valid_adjacent_inds)
+            profile_data = model.data[adjacent_condition]
+
             # Mask out bad pixels
-            profile_data = np.where(
-                model.dq[self.custom_slice(dispaxis, valid_adjacent_inds)] & self.DO_NOT_USE,
-                np.nan,
-                profile_data
-            )
-            # Add additional cut to pull only from region with valid data for convenience (may not be necessary)
-            profile_data = profile_data[self.custom_slice(3 - dispaxis, list(range(profile_cut[0], profile_cut[1])))]
+            invalid_condition = (model.dq[adjacent_condition] & self.DO_NOT_USE).astype(bool)
+            profile_data[invalid_condition] = np.nan
+
+            # Add additional cut to pull only from region with valid data
+            # for convenience (may not be necessary)
+            region_condition = self.custom_slice(3 - dispaxis, range(*profile_cut))
+            profile_data = profile_data[region_condition]
 
             # Normalize profile data
-            normalized = profile_data / np.nanmax(np.abs(profile_data), axis=(dispaxis - 1), keepdims=True)
+            # TODO: check on signs here - absolute max sometimes picks up
+            #  large negative outliers
+            profile_norm_scale = np.nanmax(np.abs(profile_data), axis=(dispaxis - 1), keepdims=True)
+            normalized = profile_data / profile_norm_scale
+
+            # Get corresponding error and variance data and scale and mask to match
+            # Handle the variance arrays as errors, so the scales match.
+            err_names = ['err', 'var_poisson', 'var_rnoise', 'var_flat']
```

Comment:
Just curious: are we interpolated new values for the output ERR array from the input ERR array, or just doing interpolation of the variances and then computing a new ERR from the new variances?